src/main/java/in/ac/bits/protocolanalyzer/persistence/repository/SaveRepository.java

```diff
@@ -1,77 +1,73 @@
+import java.util.HashMap;
 import java.util.concurrent.ConcurrentLinkedQueue;
 
+import lombok.Data;
 import lombok.extern.log4j.Log4j;
 
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.ComponentScan;
 import org.springframework.context.annotation.Scope;
 import org.springframework.data.elasticsearch.core.ElasticsearchTemplate;
 import org.springframework.data.elasticsearch.core.query.IndexQuery;
 import org.springframework.stereotype.Component;
 
-import com.google.common.eventbus.EventBus;
-import com.google.common.eventbus.Subscribe;
-
-import javax.naming.Context;
-import javax.naming.InitialContext;
-import javax.naming.NamingException;
-
-import in.ac.bits.protocolanalyzer.analyzer.event.BucketLimitEvent;
-import in.ac.bits.protocolanalyzer.analyzer.event.EndAnalysisEvent;
-import in.ac.bits.protocolanalyzer.analyzer.event.SaveRepoEndEvent;
-
 @Component
+@ComponentScan("config.in.ac.bits.protocolanalyzer.persistence.repository")
 @Scope("prototype")
 @Log4j
+@Data
 public class SaveRepository implements Runnable {
 
 	@Autowired
 	private ElasticsearchTemplate template;
-
+	
+	@Autowired
 	private ConcurrentLinkedQueue<ArrayList<IndexQuery>> buckets;
-
+	
+	@Autowired
+	private Runtime runtime;
+	
+	@Autowired
+	private HashMap<String,String> envProperties;
+	
+	private int lowWaterMark;
+	
+	private boolean analysisOnly;
+	
+	private long memory;
+	
 	private boolean isRunning = false;
 
 	private boolean analysisRunning = true;
 
 	private EventBus eventBus;
-
-	private int lowWaterMark;
-
-	private boolean analysisOnly;
-
-	public boolean isRunning() {
-		return isRunning;
-	}
-
+	
+	private static final long MEGABYTE = 1024L * 1024L;
+	
 	public void configure(EventBus eventBus) {
-		buckets = new ConcurrentLinkedQueue<ArrayList<IndexQuery>>();
 		this.eventBus = eventBus;
 		this.eventBus.register(this);
-		try {
-			Context ctx = new InitialContext();
-			Context env = (Context) ctx.lookup("java:comp/env");
-			lowWaterMark = Integer.parseInt((String) env.lookup("lowWaterMark"));
+		if(Boolean.parseBoolean(envProperties.get("noError"))) {
+			lowWaterMark = Integer.parseInt(envProperties.get("lowWaterMark"));
 			log.info("LOW WATER MARK READ FROM FILE IS: " + lowWaterMark);
-		} catch (NamingException e) {
-			log.info("EXCEPTION IN READING FROM CONFIG FILE");
-			lowWaterMark = 3;
-		}
-		//Set the value of the analysisOnly
-		try {
-			Context ctx = new InitialContext();
-			Context env = (Context) ctx.lookup("java:comp/env");
-			if (((String) env.lookup("analysisOnly")).equals("true")) {
-				analysisOnly = true;
-			} else {
-				analysisOnly = false;
-			}
+			analysisOnly = Boolean.parseBoolean(envProperties.get("analysisOnly"));
 			log.info("Perform only analysis: " + analysisOnly);
-		} catch (NamingException e) {
-			log.info("EXCEPTION IN READING FROM CONFIG FILE FOR analysisOnly .. setting false by default");
-			analysisOnly = false;
+		}
+		else {
```

Comment:
Since the SaveRepositoryConfig is guaranteeing valid values in the hashmap, I don't see a need for `envProperties.get("noError")`. You can remove the if-else structure and directly assign the received values to `lowWaterMark` and `analysisOnly` variables.